### DISCUSSION TITLE
Update coding guidelines (documenting new scripting functions)

### DISCUSSION THREAD

**Discussion by Dutchman101** [multitheftauto/mtasa-blue#1938] [Ideas]
**Idea**

We should formalize that PR's with new scripting functions won't be merged, until the PR author created a scripting wiki page for it.. those pages should be separated from other content (not on overviews and tagged `WIP`) until date of merge.

For a long time it's been said to be, and loosely applied as, coding guidelines but it's missing from the [coding guidelines](https://github.com/multitheftauto/mtasa-blue/blob/master/CONTRIBUTING.md) so after working it out, we should add it there so there'll be no misunderstandings and no growing list of undocumented functions.

The past has learned that telling contributors adding the wiki page is their responsibility, won't always cut it.. sometimes it won't be done, or at least not before someone else random takes interest in doing it. It's a shame, because we risk entire functions going to waste because without documentation on the scripting wiki, no one will use them.

After implementing such a policy change, we should change our Github practises to reflect it - bottom line will be making it a check that is required to complete before a PR can be merged. So we could do one of the following:

- Do it manually, post a request for completion and state it won't be merged otherwise
- Use a tag like "documentation-missing" which is a blocker for merge (probably the best option)
- Use a bot that automatically verifies the existence of a wiki page according to the function's naming (unlikely)

Even if documenting their new function is primarily the PR author's responsibility, obviously anyone else that likes to do it can still take care of it for them. But also then, it simply has to happen in advance of the merge, to avoid problems of there being no prospect of it being done.

If this discussion leads to a policy update, afterwards it can also be worth running an automated script (once) to get all scripting functions that exist from MTA sourcecode, and then verifies if all of them are on the wiki. So we can catch up to accumulated undocumented functions from the past.

**Comment by Citizen01** [comment]
I agree on that, but I don't like the fact a wiki page exists for a non existing (yet) function.
Because you can't even put a `BETA: FROM VERSION 1.X.Y rZZZZZ ONWARDS` because this version doesn't exist.
Or else we need a new thing/template to clearly state this function is still at the PR stage (with a direct link to the PR, would be cool)

**Reply by Dutchman101** [reply]
> I agree on that, but I don't like the fact a wiki page exists for a non existing (yet) function.
> Because you can't even put a `BETA: FROM VERSION 1.X.Y rZZZZZ ONWARDS` because this version doesn't exist.
> Or else we need a new thing/template to clearly state this function is still at the PR stage (with a direct link to the PR, would be cool)

I wrote about it at the start:
`those pages should be separated from other content (not on overviews and tagged WIP) until date of merge.`

Besides not having it in the scripting function overviews and indexes, which probably won't lead any scripters there.. we can intepret "tagged WIP" as having an unmissable warning box (template) that said function doesn't exist yet

> clearly state this function is still at the PR stage (with a direct link to the PR, would be cool)

Good ideas for the template

**Comment by Citizen01** [comment]
> I wrote about it at the start

Ups yeah, I've somehow skipped it, my bad.
